### PR TITLE
feat: include `X-Ably-ClientId` for each request (RSA7e2)

### PR DESCRIPTION
**Pull request by ttypic**
Resolves https://github.com/ably/ably-java/issues/1015

### CODE REVIEW

**Review by sacOO7 — APPROVED** [review]
Lgtm
You can check for failing tests though

File: lib/src/test/java/io/ably/lib/test/rest/HttpHeaderTest.java [hunk 1]
```diff
@@ -83,12 +83,50 @@ public void header_lib_channel_publish() {
...
+            /* Publish message */
+            String messageName = "test message";
+            String messageData = String.valueOf(System.currentTimeMillis());
+
+            Channel channel = ably.channels.get("test");
+            channel.publish(messageName, messageData);
+
+            /* Get last headers */
+            Map<String, String> headers = server.getHeaders();
+            Assert.assertEquals(headers.get("x-ably-clientid"), "dGVzdCBjbGllbnQ=");
```
Comment (sacOO7) [inline lib/src/test/java/io/ably/lib/test/rest/HttpHeaderTest.java:123]:
```suggestion
            Assert.assertEquals(headers.get("x-ably-clientid"),Base64Coder.encodeString("test client"));
```

**Comment by ttypic** [comment]
thanks @sacOO7, yeah, looking at it right now, oddly it passes locally

File: lib/src/test/java/io/ably/lib/test/rest/HttpHeaderTest.java (hunk 1 above)
Comment (ttypic) [reply lib/src/test/java/io/ably/lib/test/rest/HttpHeaderTest.java:123]: I prefer not to put logic for expected value. Potential bug in `Base64Coder#encodeString` method can make test unreliable

Comment (ttypic) [reply lib/src/test/java/io/ably/lib/test/rest/HttpHeaderTest.java:123]: I'll add comment that it is in fact base64 encoded "test client" value, to make it more readable

Comment (sacOO7) [reply lib/src/test/java/io/ably/lib/test/rest/HttpHeaderTest.java:123]: Sure, no issues. Just that, the assertion was more readable. Agree that assertions should be dumb and shouldn't be any logic into them.

Comment (sacOO7) [reply lib/src/test/java/io/ably/lib/test/rest/HttpHeaderTest.java]: lg

**Pull request merged**